tests/tests_integration/test_api/test_functions.py

```diff
@@ -16,3 +16,13 @@ def test_retrieve_unknown_raises_error(self, cognite_client: CogniteClient):
     def test_retrieve_unknown_ignore_unknowns(self, cognite_client: CogniteClient):
         res = cognite_client.functions.retrieve_multiple(external_ids=["this does not exist"], ignore_unknown_ids=True)
         assert len(res) == 0
+
+    def test_function_list_schedules_unlimited(self, cognite_client: CogniteClient):
+        expected_unique_schedules = 5
+        # This is an integration test dummy function that purposefully doesn't have an external id.
+        fn = cognite_client.functions.retrieve(id=2495645514618888)
+        schedules = fn.list_schedules(limit=-1)
+
+        assert len(schedules) == expected_unique_schedules
+        assert len({s.id for s in schedules}) == expected_unique_schedules
+        assert len({s.cron_expression for s in schedules}) == expected_unique_schedules
```

Comment:
This test now catches the duplication error. I created other dummy functions in the project with schedules, with and without external id.

Tbh, I think the behaviour of the schedules/list is not what I'd expect if external_id = None. I though it'd return an empty list 🤷‍♂️